utils/sparse_mt.py

```diff
@@ -0,0 +1,432 @@
+    agg_expr['FS'] = fs_from_sb(agg_expr['SB'])
+
+    # Run aggregator on non-ref genotypes
+    info = hl.agg.filter(
+        mt.LGT.is_non_ref(),
+        hl.struct(
+            **{k: v for k, v in agg_expr.items() if k != 'DP'}
+        )
+    )
+
+    # Add DP, computed over both ref and non-ref genotypes, if present
+    if 'DP' in agg_expr:
+        info = info.annotate(
+            DP=agg_expr['DP']
+        )
+
+    return info
+
+
+def impute_sex_ploidy(
+        mt: hl.MatrixTable,
+        excluded_intervals: Optional[hl.Table],
+        normalization_contig: str = 'chr20',
+        chr_x: Optional[str] = None,
+        chr_y: Optional[str] = None,
+) -> hl.Table: # TODO: For exomes, calling intervals need to be added
+    """
+    Imputes sex ploidy from a sparse Matrix Table by normalizing the coverage of chromosomes X and Y using
+    the coverage of an autosomal chromosome (by default chr20).
+    Coverage is computed using the median block coverage (summed over the block size) and the non-ref coverage at non-ref genotypes.
+
+    :param MatrixTable mt: Input sparse Matrix Table
+    :param Table excluded_intervals: An optional table of intervals to exclude from the computation.
+    :param str normalization_contig: Which chromosome to normalize by
+    :param str chr_x: Optional X Chromosome contig name (by default uses the X contig in the reference)
+    :param str chr_y: Optional Y Chromosome contig name (by default uses the Y contig in the reference)
+    :return: Table with mean coverage over chromosomes 20, X and Y and sex chromosomes ploidy based on normalized coverage.
+    :rtype: Table
+    """
+
+    ref = get_reference_genome(mt.locus, add_sequence=True)
+    if chr_x is None:
+        if len(ref.x_contigs) != 1:
+            raise NotImplementedError(
+                "Found {0} X chromosome contigs ({1}) in Genome reference. sparse_impute_sex_ploidy currently only supports a single X chromosome contig. Please use the `chr_x` argument to  specify which X chromosome contig to use ".format(
+                    len(ref.x_contigs),
+                    ",".join(ref.x_contigs)
+                )
+            )
+        chr_x = ref.x_contigs[0]
+    if chr_y is None:
+        if len(ref.y_contigs) != 1:
+            raise NotImplementedError(
+                "Found {0} Y chromosome contigs ({1}) in Genome reference. sparse_impute_sex_ploidy currently only supports a single Y chromosome contig. Please use the `chr_y` argument to  specify which Y chromosome contig to use ".format(
+                    len(ref.y_contigs),
+                    ",".join(ref.y_contigs)
+                )
+            )
+        chr_y = ref.y_contigs[0]
+
+    def get_contig_size(contig: str) -> int:
+        contig = hl.utils.range_table(ref.contig_length(contig), n_partitions=int(ref.contig_length(contig) / 500_000))
+        contig = contig.annotate(
+            locus=hl.locus(contig=contig, pos=contig.idx + 1)
```

Comment:
maybe it's a good idea to add `reference_genome=ref`? otherwise this assumes the default reference genome matches whichever build the mt is in

Reply:
good catch!